### PR TITLE
Add Hold/Recall Form

### LINKED ISSUE #142: Add Hold/Recall Form

### PR DESCRIPTION
**Pull request by jkeck**
Closes #142

Example record:
`/requests/new?barcode=36105212925395&item_id=10785714&origin=GREEN&origin_location=STACKS`

hold-recall-form

### CODE REVIEW

File: app/models/request.rb [hunk 1]
```diff
@@ -80,8 +85,9 @@ def find_existing_user
   end
 
   def holdings
-    if persisted?
-      @holdings ||= searchworks_item.requested_holdings.by_barcodes(barcodes)
+    if persisted? || requested_barcode.present?
```
Comment (jkeck) [inline app/models/request.rb]: This is just temporary, I would like to find a better way of doing this in general.

File: app/views/hold_recalls/_header.html.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+<h1 id='dialogTitle'>Request [checked-out | on-order] item</h1>
```
Comment (jkeck) [inline app/views/hold_recalls/_header.html.erb]: I need to update this.  We don't have the tooling to know if the item is one or the other, so I need to build that in.

File: app/controllers/mediated_pages_controller.rb [hunk 3]
```diff
@@ -27,11 +27,12 @@ def validate_mediated_pageable
     fail UnmediateableItemError unless @mediated_page.mediateable?
   end
 
+  # this should not be page param (needs test)
```
Comment (cbeer) [inline app/controllers/mediated_pages_controller.rb]: Is this a TODO, or a ticket that should be made?

Comment (jkeck) [inline app/controllers/mediated_pages_controller.rb]: Good catch. Thanks! This was actually addressed in 215035ebbfb39c8af2052cfdc520cef2ee1a2417 but I neglected to remove the comment.  Just took that out and pushed the branch back up.

**Comment by cbeer** [comment]
:+1:

**Pull request merged**